prebid-server/pbs-modules/rules-engine.md

```diff
@@ -0,0 +1,805 @@
+                            "analyticsValue": "leaf6"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","web","false","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderD","bidderF"],
+                            "analyticsValue": "leaf7"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","*","*","true"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderC"],
+                            "analyticsValue": "leaf8"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","*","*","false"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderF"],
+                            "analyticsValue": "leaf9"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","web","*","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": [],
+                            "analyticsValue": "leaf10"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","*","true","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderB", "bidderE"],
+                            "analyticsValue": "leaf11"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","*","false","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderE", "bidderF"],
+                            "analyticsValue": "leaf12"
+                        }]
+                    }]
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+And here's how these scenarios would play out in various runtime conditions:
+
+1. **Input**: request from country FRA, channel web, contains eid(pubcid.org), userFpd is available
+
+   - results: resolves to leaf 4  
+   - remove bidderC from imp\[0\] and imp\[1\]  
+   - aTag logs leaf4
+
+2. **Input**: request from country JPN, contains eid(pubcid.org), userFpd is available
```

Comment:
"2. **Input**: request from country JPN, channel web, contains eid(pubcid.org), userFpd is available"